src/mrinufft/operators/interfaces/torchkbnufft.py

```diff
@@ -0,0 +1,281 @@
+        Shape of the image space.
+    density : bool or Tensor, optional
+        Density compensation support. Default is False.
+        - If a Tensor, it will be used for density.
+        - If True, the density compensation will be automatically estimated
+            using the fixed point method.
+        - If False, density compensation will not be used.
+    n_coils : int, optional
+        Number of coils. Default is 1.
+    n_batchs : int, optional
+        Number of batches. Default is 1.
+    smaps : Tensor, optional
+        Sensitivity maps. Default is None.
+    eps : float, optional
+        A small epsilon value for numerical stability. Default is 1e-6.
+    squeeze_dims : bool, optional
+        If True, tries to remove singleton dimensions for batch and coils.
+        Default is True.
+    use_gpu : bool, optional
+        Whether to use the GPU. Default is False.
+    osf : int, optional
+        Oversampling factor. Default is 2.
+    **kwargs : dict
+        Additional keyword arguments.
+
+    """
+
+    available = TORCH_AVAILABLE
+    autograd_available = False
+
+    def __init__(
+        self,
+        samples,
+        shape,
+        density=False,
+        n_coils=1,
+        n_batchs=1,
+        smaps=None,
+        eps=1e-6,
+        squeeze_dims=True,
+        use_gpu=False,
+        osf=2,
+        **kwargs,
+    ):
+        super().__init__()
+
+        self.use_gpu = use_gpu
+        self.shape = shape
+        if isinstance(samples, torch.Tensor):
+            if is_cuda_tensor(samples):
+                samples = samples.cpu()
+            samples = samples.numpy()
+        samples = proper_trajectory(
+            samples.astype(np.float32, copy=False), normalize="pi"
+        )
+        self.samples = torch.tensor(samples).to("cuda" if use_gpu else "cpu")
+        self.dtype = None
+        # self.dtype = self.samples.dtype
+        self.n_coils = n_coils
+        self.n_batchs = n_batchs
+        self.squeeze_dims = squeeze_dims
+        # self.eps = eps
+        self.compute_density(density)
+        self.compute_smaps(smaps)
+        if isinstance(smaps, np.ndarray) or (
+            CUPY_AVAILABLE and isinstance(smaps, cp.ndarray)
+        ):
+            self.smaps = torch.tensor(smaps).to("cuda" if use_gpu else "cpu")
+
+        self._tkb_op = tkbn.KbNufft(im_size=self.shape).to("cuda" if use_gpu else "cpu")
+        self._tkb_adj_op = tkbn.KbNufftAdjoint(im_size=self.shape).to(
+            "cuda" if use_gpu else "cpu"
+        )
+
+    @with_torch
+    def op(self, data, out=None):
+        """Forward operation.
+
+        Parameters
+        ----------
+        data: Tensor
+
+        Returns
+        -------
+        Tensor: Non-uniform Fourier transform of the input image.
+        """
+        B, C, XYZ = self.n_batchs, self.n_coils, self.shape
+        data = data.reshape((B, 1 if self.uses_sense else C, *XYZ))
+        if self.use_gpu:
+            data = data.to("cuda", copy=False)
```

Comment:
to avoid the various if checks, I would recommend instead:
- a `device` attribute in the init (in place of use_gpu). That could only be "cpu" or "cuda" (or a torch.device object). this also leaves the door open for multi GPU support.